### DISCUSSION TITLE
Is it possible to use direct mapping?

### DISCUSSION THREAD

**Discussion by lambdakris** [morph-kgc/morph-kgc#108] [Q&A] [answered]
Hi, I was wondering if it was possible to use Direct Mapping with any of the supported data sources?

**Comment by arenas-guerrero-julian** [comment, marked as answer]
Hi @lambdakris ,

As far as I know direct mapping is only possible for relational databases. Unfortunately, morph-kgc does not support direct mapping at this moment.

**Comment by lambdakris** [comment]
Thanks for clarifying @ArenasGuerreroJulian. Along those lines, I am curious whether you have any insight to the relationship between morph-kgc and morph-rdb (when might you use one and not the other). The reason I ask is because I was reading the paper [MIRROR: Automatic R2RML Mapping Generation from Relational Databases](https://link.springer.com/content/pdf/10.1007/978-3-319-19890-3_21.pdf) and the authors mention contributing support for generating rdb2rml mappings that conform to the direct mapping algoritm to the morph-rdb project.

**Comment by arenas-guerrero-julian** [comment]
morph-kgc is a materialization engine, that is, you the entire RDF ffrom a relational database. You may then exploit that RDF for instance by loading it to a triplestore such as [Virtuoso](https://github.com/openlink/virtuoso-opensource).

morph-rdb is a vitualization engine, that is it translates SPARQL queries into SQL queries, and produces the result set on the fly.
